simulation/simple_sensor_simulator/test/src/sensor_simulation/lidar/test_lidar_sensor.hpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SIMPLE_SENSOR_SIMULATOR__TEST__TEST_LIDAR_SENSOR_HPP_
+#define SIMPLE_SENSOR_SIMULATOR__TEST__TEST_LIDAR_SENSOR_HPP_
+
+#include <gtest/gtest.h>
+#include <simulation_api_schema.pb.h>
+
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <set>
+#include <simple_sensor_simulator/sensor_simulation/lidar/lidar_sensor.hpp>
+#include <string>
+#include <traffic_simulator_msgs/msg/entity_status.hpp>
+#include <vector>
+
+using namespace simple_sensor_simulator;
+
+class LidarSensorTest : public ::testing::Test
+{
+protected:
+  LidarSensorTest()
+  {
+    configureLidarSensor();
+    initializeEntityStatuses();
+  }
+
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+    node_ = std::make_shared<rclcpp::Node>("lidar_sensor_test_node");
+    createRosInterface();
+
+    lidar_ = std::make_unique<LidarSensor<sensor_msgs::msg::PointCloud2>>(0.0, config_, publisher_);
+  }
+  void TearDown() override { rclcpp::shutdown(); }
+
+  rclcpp::Node::SharedPtr node_;
+  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr publisher_;
+  rclcpp::Subscription<sensor_msgs::msg::PointCloud2>::SharedPtr subscription_;
+
+  traffic_simulator_msgs::EntityStatus ego_status_;
+  traffic_simulator_msgs::EntityStatus other1_status_;
+  traffic_simulator_msgs::EntityStatus other2_status_;
+  std::vector<traffic_simulator_msgs::EntityStatus> status_;
+
+  std::unique_ptr<LidarSensorBase> lidar_;
+  simulation_api_schema::LidarConfiguration config_;
+  sensor_msgs::msg::PointCloud2::SharedPtr received_msg_;
+
+  double current_simulation_time_{1.0};
+  rclcpp::Time current_ros_time_{1};
+
+private:
+  auto configureLidarSensor() -> void
```

Comment:
Have you seen `traffic_simulator::helper::constructLidarConfiguration` function? maybe it could be helpful

Reply:
maybe it is better to add a new directory 'utils' and put there helper_functions.hpp with defintion of constructLidarConfiguration